packages/plugin-constraints/sources/Constraints.ts

```diff
@@ -18,6 +19,34 @@ const DEPENDENCY_TYPES = [
   DependencyType.PeerDependencies,
 ];
 
+function clearerError(tauErr: any) {
+  let prologCode = String(tauErr);
+  prologCode = prologCode.replace(/^throw\(/g, `_throw(`);
+  prologCode = prologCode.replace(/'\/'\(([^,]+)/g, `slashfn('$1'`);
+
+  const clearerErr = runInNewContext(prologCode, {
+    [`_throw`]: (err: Error) => err,
+    [`error`]: (err: Error, args: Array<any>) => Object.assign(err, ... args),
+    [`syntax_error`]: (reason: string) => new ReportError(MessageName.PROLOG_SYNTAX_ERROR, `Syntax error: ${reason}`),
+    [`existence_error`]: (type: string, description: string) => new ReportError(MessageName.PROLOG_EXISTENCE_ERROR, `Existence error: ${type} ${description} doesn't exist`),
+    [`slashfn`]:  (name: string, arity: number) => `${name}/${arity}`,
+    [`line`]: (line: number) => ({line}),
+    [`column`]: (column: number) => ({column}),
+    [`procedure`]: `procedure`,
+    [`top_level`]: `top level`,
+    [`found`]: () => ({}),
+    [`token_not_found`]: ({}),
+  });
```

Comment:
I guess this is the elephant in the room 🤔 It's pretty ugly but it gets the job done without requiring a whole new parser.

Reply:
Couldn't live with this horrible thing, I made a basic interpreter that does the same trick more cleanly (no potential risk of remote code evaluation).